content/05-mysql/10-reading-and-querying-data/05-optimizing-slow-queries.mdx

```diff
@@ -0,0 +1,446 @@
+
+One thing that might negatively affect your query's performance is over-reliance on wildcards.  In SQL, the `%` character is used in comparisons as a wildcard indicating that any value can be substituted.  This is a very powerful way of finding data within your records, but it can have very big impacts on query performance.
+
+This is especially true for leading or prefix wildcards.  Leading wildcards are comparisons that check whether a column value matches a pattern that begins with a wildcard.  For instance, checking whether a column matches `%Main St.`.  This performs especially bad because MySQL is not able to use indexes to discover matching results.  Instead, it must look through the entire table for matches, even if that column has an index.
+
+In this particular example, if you will be regularly querying by a street name to discover all addresses on that street, it might make more sense to split up the street name and street numbers into two separate columns.  This would allow you to query for addresses on "Main St." using an index quickly and then filter those results as necessary if you are looking for a specific address.  Creating a compound index with the street name first and the street number column second would make this even faster.
+
+## Conclusion
+
+In this guide, we walked through some of the basic ways of evaluating individual queries and getting more information about what might be slowing them down.  We discussed profiling queries and using MySQL's `EXPLAIN` statement to get detailed information about how the query planner evaluates the query.  We then took a look at some ways that you can avoid these slow downs by modifying your queries to align better with the way the database retrieves data.
+
+Databases are complex pieces of software that are responsible for operating on large amounts of data that can be in many different formats.  In addition to simply retrieving the necessary data, there's an expectation that the database should be able to return results quickly.  There are countless optimizations built into MySQL's query planner and optimizer mechanisms to help minimize the amount of time it takes to answer queries.
+
+However, the database is only able to automatically optimize in a limited sense.  As the user of the database, you need to also be able to understand the data being accessed, your requirements for a query, and which queries are most important to optimize for.  For this, it is important to understand how to evaluate your criteria and make adjustments when the query performance is not at the expected level.
+
+<PrismaOutlinks>
+
+If you are using Prisma with your MySQL database, you can about ways to optimize your queries in the [query optimization section of the docs](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance).  This will help you understand how various query constructions can impact your database performance when using Prisma.
```

Comment:
```suggestion
If you are using Prisma with your MySQL database, you can read about ways to optimize your queries in the [query optimization section of the docs](https://www.prisma.io/docs/guides/performance-and-optimization/query-optimization-performance).  This will help you understand how various query constructions can impact your database performance when using Prisma.
```